feat: :art: adding authentication and error handeling

Rebal67 opened the pull request.

theblockstalk reviewed (commented):
how do you know this is running correctly? i don't see any tests for this

Rebal67 commented:
> how do you know this is running correctly? i don't see any tests for this



> how do you know this is running correctly? i don't see any tests for this

Fixed Tests and added one for fail case

theblockstalk commented:
:tada: This PR is included in version 1.0.0-development.1 :tada:

The release is available on [GitHub release](https://github.com/Tonomy-Foundation/Tonomy-Communication/releases/tag/1.0.0-development.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

theblockstalk commented:
:tada: This PR is included in version 1.0.0-rc.1 :tada:

The release is available on [GitHub release](https://github.com/Tonomy-Foundation/Tonomy-Communication/releases/tag/1.0.0-rc.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
